notebooks/test_idata_general_time_representation.py

```diff
@@ -0,0 +1,237 @@
+"""
+Experiment file for testing
+general representation of timepoints
+in idata object (intervals not just
+1D).
+"""
+
+# %% LIBRARY IMPORTS
+
+from datetime import datetime, timedelta
+
+import arviz as az
+import polars as pl
+
+import forecasttools
+
+# %% LOAD IDATA WO DATES
+
+idata_wo_dates = forecasttools.nhsn_flu_forecast_wo_dates
+
+# print(idata_wo_dates["observed_data"]["obs_dim_0"])
+
+# print(idata_wo_dates["posterior_predictive"])
+
+
+# %% FUNCTION FOR MODIFYING SINGLE VARIABLE'S DIMENSIONS
+
+
+def add_time_coords_to_idata_variable(
+    idata: az.InferenceData,
+    group: str,
+    variable: str,
+    dimensions: list[str],
+    start_date_iso: str,
+    time_step: timedelta,
+) -> az.InferenceData:
+    """
+    Adds time coordinates to a specified variable
+    within a group in an ArviZ InferenceData object.
+    This function assigns a range of time coordinates
+    to a specified dimension in a variable
+    within a group in an InferenceData object.
+
+    Parameters
+    ----------
+    idata : az.InferenceData
+        The InferenceData object containing
+        the group and variable to modify.
+    group : str
+        The name of the group within the InferenceData
+        object (e.g., "posterior_predictive").
+    variable : str
+        The name of the variable within the
+        specified group to assign time coordinates to.
+    dimensions : list[str]
+        A list of dimension names to which time
+        coordinates should be assigned.
+    start_date_iso : str
+        The start date for the time coordinates
+        in ISO format (e.g., "2022-08-08").
```

Comment:
accept either a string or a `datetime.date`?

Reply:
Or even just a `datetime.date` if you're going to require the next argument to be a `timedelta` (as opposed to, e.g., a parseable string)

Reply:
Does the option (`start_date_iso` as `str` or `datetime.date` AND `time_steps` as `str` or `datetime.timedelta`) seem better to you than (`start_date_iso` as `datetime.date` and `time_steps` `datetime.timedelta`)?

Reply:
I think my preference is `start_date` as `str` or `datetime.date` and time_step as `datetime.timedelta` is my preference

Reply:
Sounds good; I will go with this preference.